src/test/kotlin/org/rust/ide/refactoring/RsInlineFunctionTest.kt

```diff
@@ -866,11 +867,17 @@ class RsInlineFunctionTest : RsTestBase() {
                 let mut quux = 5;
                 return &mut quux;
             }
+
+            fn corge() -> &mut i32 {
```

Comment:
Could you also please make the test code compilable with rustc? It could be a bit hard with `&mut` references (although you could use &'static refs and some unsafe code), maybe choose some simpler code snippet?

Reply:
I can make `corge()` return a plain `i32` that will have a mutable binding later and be done with that part, but I do need to handle the errors in `qux()`, where a mutable reference is the point. 
I'll try to handle the lifetimes as the compiler suggests, but it might take me some time.

Reply:
Well, I have a piece of test code that passes rustc. However, it doesn't pass our own checks, saying `Cannot move` under `*i` and `*j`
```rust
fn main() {
    let _a = foo(&bar(), &mut qux(), corge());
}

#[allow(unused_mut)]
fn /*caret*/foo(i: &i32, j: &mut i32, mut k: i32) -> i32 {
    return *i + *j + k + bar();
}

fn bar() -> i32 {
    let baz = 5;
    return baz;
}

fn qux() -> i32 {
    let quux = 5;
    return quux;
}

fn corge() -> i32 {
    let grault = 5;
    return grault;
}
```

Reply:
Try to add `@ProjectDescriptor(WithStdlibRustProjectDescriptor::class)` above the test function. I think that it needs stdlib to see macros that implement `Copy` for primitive types.

Reply:
It doesn't help unfortunately, both because I'm using *Inject Language* so it already knows it's *Rust*, and because I tried it all in a scratch file first, so there it's pure *Rust*. I guess that it's a bug in the plugin.

Reply:
I checked out your branch and the test passes, so it's OK. This code also works in the plugin if you use it inside some project.

Reply:
Did you use in the tests what I commited or the code I supplied above? because I only committed it now, the test doesn't pass on my machine, and if it didn't work in a scratch file it shouldn't work in a project either (unless cargo related).

Reply:
Sorry, didn't see the last commit. But still, the code works inside a project. It needs access to a project and to stdlib to infer `Copy` properly. Inside injected Rust strings and scratch files it does not have access to a project, that's why it shows the `Cannot move` false positive.

![image](https://user-images.githubusercontent.com/4539057/94438430-62908b80-019f-11eb-95a6-fef83228f9df.png)

Reply:
That's good, means the code itself, doesn't need change any more.
I also see from searching that `@ProjectDescriptor(WithStdlibRustProjectDescriptor::class)` is the standard way to introduce `stdlib` inside, any idea why it's not working?

Reply:
If the problem that you're trying the solve is that there are red annotations inside the injected Rust test code, then the project descriptor will not solve it. But it's not a problem by itself, such annotations are present in a lot of tests, it's completely fine. Do you have any other problems with this test?

Reply:
No, I just assumed there is a way to solve it 😅